### PR TITLE
Remove Cats Effect dependency from core

### PR DESCRIPTION
**Pull request by andreamarcolin**
Apologies if it took me this much time to get round to submit this PR, busy times! 🤷🏻‍♂️ 

This is an attempt at freeing core from the cats-effect dependency, see #156.
I went for two separate traits as opposed to just one because it was the quickest thing to do, so that I didn't have to touch `UUIDProvider` but just replace `Clock` with `TimeProvider`. 
I also moved the `Sync` instance for `UUIDProvider` to the http4s-emitter module, and moved `Id` instances to the id-emitter module accordingly. 
Most of these are breaking changes, but I don't see a way out of this without breaking compatibility. 
Metadata module was left unchanged, as @chuwy said in the issue, it sucks but not much we can do probably.

If this is a step in the right direction, I will then follow-up with a CE2 -> CE3 PR for the metadata and http4s-emitter modules.

Merry Christmas and happy new Year everyone!

### CODE REVIEW

**Comment by snowplowcla** [comment]
Thanks for your pull request. Is this your first contribution to a Snowplow open source project? Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).

:memo: Please visit https://docs.snowplowanalytics.com/docs/contributing/contributor-license-agreement/ to learn more and sign.

Once you've signed, please reply here (e.g. I signed it!) and we'll verify. Thanks.

File: modules/http4s-emitter/src/main/scala/com/snowplowanalytics/snowplow/scalatracker/emitters/http4s/package.scala [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package com.snowplowanalytics.snowplow.scalatracker.emitters
```
Comment (chuwy) [inline modules/http4s-emitter/src/main/scala/com/snowplowanalytics/snowplow/scalatracker/emitters/http4s/package.scala:13]: Missing license header.

File: modules/id-emitter/src/main/scala/com/snowplowanalytics/snowplow/scalatracker/emitters/id/package.scala [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package com.snowplowanalytics.snowplow.scalatracker.emitters
```
Comment (chuwy) [inline modules/id-emitter/src/main/scala/com/snowplowanalytics/snowplow/scalatracker/emitters/id/package.scala:13]: Missing license header.

**Review by chuwy — COMMENTED** [review]
I quite like how it turned out.

>  Most of these are breaking changes, but I don't see a why out of this without breaking compatibility.

I think that's completely fine as long as we use appropriate versioning.

The only suggestion I have is to still combine `TimeProvider` and `UUIDProvider` into a single trait (`Tracking`? `Tracker.Capability`? Not sure, tbh). The reasoning is that:

* Most of the time, in user codebase they will appear together and long context-bounds are very annoying
* Both of them have exact counterparts in CE3 and having two sets of identical algebras seems a bit wasteful
* The less entities - the better
* Last, but not least - we're breaking compatibility anyways, so perhaps it's the good time

**Comment by andreamarcolin** [comment]
@chuwy thanks for the quick review! I addressed your concerns, went with Tracking since I couldn't come up with anything better 😃

**Review by chuwy — APPROVED** [review]
Awesome!

**Pull request merged**